### PR TITLE
chore(test): fix determining server version when no env var + local server present MONGOSH-416

### PR DESCRIPTION
**Pull request by rose-m**
I broke the `compile_and_upload` part in #496 inside docker. This fixes it.

### CODE REVIEW

File: testing/integration-testing-hooks.ts
Comment (mcasimir) [inline testing/integration-testing-hooks.ts:472]: this runs a `before` inside a `before`, how is it going to work?

File: testing/integration-testing-hooks.ts
Comment (mcasimir) [inline testing/integration-testing-hooks.ts:468]: What is the 'live' version here :)?

Starting a mongod server just for the version probably is unnecessary.

My understanding of how the tests are setup here is that mlaunch without an env var would:

1. try to see if a mongod binary is in the path an use it
2. if not download and start the latest

In that case maybe we could do something like this (super-inaccuarte-pseudo-code):

``` js
const ver = await commandExists('mongod')? parseVersion(await exec('mongod --version')) : 9999.9999.9999
```

Comment (rose-m) [reply testing/integration-testing-hooks.ts:468]: The "live" version is the one we really end up using :D But good point I could default to a super high version number since it is the "latest".

Shutdown is not necessary as it is a "shared" instance and will be re-used and later shut down automatically.

File: testing/integration-testing-hooks.ts
Comment (rose-m) [reply testing/integration-testing-hooks.ts:472]: Good question 😅 It did work locally tho - I'd assume since it's still in "before" the other before hook gets registered and executed. With your suggestion below it would be a lot simpler and not required to launch it here.

**Review by mcasimir — CHANGES_REQUESTED** [review]
Had no idea such a corner case existed! Cool that you spotted it. Left some comments maybe we can avoid starting a new server for detecting the version (if i understood the use case correctly .. which i may not :))

File: testing/integration-testing-hooks.ts
Comment (mcasimir) [reply testing/integration-testing-hooks.ts:468]: Oh nice!

File: testing/integration-testing-hooks.ts
Comment (mcasimir) [reply testing/integration-testing-hooks.ts:472]: Obscure mocha voodoo, well if it works .. :)

File: testing/integration-testing-hooks.ts
Comment (mcasimir) [reply testing/integration-testing-hooks.ts:468]: > since it is the "latest"
Oh is always latest? When i run it locally it uses the mongod in the path, which i have as 4.0. 

Anyway if this approach works better then we can also merge that.

Comment (rose-m) [reply testing/integration-testing-hooks.ts:468]: Yeah _if_ it finds one on the path, mlaunch will use that. Otherwise it downloads `latest`.

But that's unfortunately exactly the corner case: there is nothing on the path and nothing in the env vars. So we "dynamically" end up with what gets downloaded for `latest`. So in the case here it will work even with the hard-coded "9999.9999.9999" version since I'm comparing against a "past version". I'll hard code that version and put it in the comment since needing this function is already an edge case by itself 😅 
We only need this since the server parameters changed from one version to the other inside our testing matrix...

Comment (mcasimir) [reply testing/integration-testing-hooks.ts:468]: I see ok!

Comment (mcasimir) [reply testing/integration-testing-hooks.ts:468]: Ping me if you need another review, going to pre-approve it so you can merge if i'm not around

**Review by mcasimir — APPROVED** [review]

**Pull request merged**
